graphx/src/main/scala/org/apache/spark/graphx/lib/StronglyConnectedComponents.scala

```diff
@@ -64,11 +69,20 @@ object StronglyConnectedComponents {
         // write values to sccGraph
         sccGraph = sccGraph.outerJoinVertices(finalVertices) {
           (vid, scc, opt) => opt.getOrElse(scc)
-        }
+        }.cache()
+        // take triplet to materialize vertices and edges (faster then count on both)
+        sccGraph.triplets.take(1)
```

Comment:
I wasn't sure about this take to be honest. I checked both solutions - with take on triplets and with count on vertices and edges separately. After return from scc count on vertices and edges showed it was generally faster with take, but it could be due to small amount of partitions. 
Count on both vertices and edges with randomly generated graph of 500k vertices and 2mln vertices adds about 20s to scc, but may be worth it (didn't show during my tests but I guess you could be right). It could also be that one partition of triplet depends on whole verticesRDD and edgesRDD so, it would be sufficient to make take(1).